device/core/src/internal_client.ts

```diff
@@ -92,8 +92,7 @@ export abstract class InternalClient extends EventEmitter {
       }
       if (err && this._retryPolicy.shouldRetry(err)) {
         /*Codes_SRS_NODE_INTERNAL_CLIENT_16_098: [If the transport emits a `disconnect` event while the client is subscribed to direct methods the retry policy shall be used to reconnect and re-enable the feature using the transport `enableMethods` method.]*/
-        if (this._methodsEnabled) {
-          this._methodsEnabled = false;
+        if (this._userRegisteredMethodListener) {
```

Comment:
We want to re-enable methods if the user registered a listener.
Before, we were enabling if the user registered a listener AND the transport successfully enabled methods prior.